exercises/practice/rest-api/src/RestApi.elm

```diff
@@ -0,0 +1,36 @@
+module RestApi exposing (buildDatabase, get, post)
+
+import Dict exposing (Dict)
+import Json.Decode exposing (Error)
+import Json.Encode
+
+
+type alias Name =
+    String
+
+
+type alias User =
+    { name : Name
+    , owes : Dict Name Int
+    , owedBy : Dict Name Int
+    , balance : Int
+    }
+
+
+type alias Database =
+    Dict Name User
+
+
+buildDatabase : String -> Result Error Database
+buildDatabase payload =
+    Debug.todo "Please implement buildDatabase"
+
+
+get : Database -> String -> Maybe String -> String
+get database url maybePayload =
+    Debug.todo "Please implement get"
+
+
+post : Database -> String -> String -> String
+post database url payload =
```

Comment:
This is a bit easier to work out, as there is only one response in the instructions table, but in terms of making the exercise more approachable (and more about working with JSON, and less about working out what the exercise is asking) it also might be worth making things easier.

Reply:
I thought about this a bunch, and I don't really want to simplify the exercise so that it's only about JSON decoding. That sounds like something a concept exercise should be doing, so I will leave this one the way it is and start thinking about a good concept exercise.

It's unfortunate that this one is really the only suitable practice exercise for JSON decoding but it's understandable. Most languages will use some kind of library and not think about it, Elm really stands out in that aspect, which is why this is one of the hardest concepts for newcomers.